src/openads-appnexus/repository/PullingDataEntry.js

```diff
@@ -0,0 +1,44 @@
+const TIMEOUT_EXCEPTION = 'Timeout retrieving the Ad from the server'
+
+export default class PullingDataEntry {
+  constructor ({id, adRepository, timeout, wait}) {
+    this._id = id
+    this._adRepository = adRepository
+    this._timeout = timeout
+    this._wait = wait
+    this._dataPromise = Promise.race([
+      this._createIntervalPromise({id}),
+      this._createTimeoutPromise()
+    ])
+  }
+
+  get id () {
+    return this._id
+  }
+
+  waitForData () {
+    return this._dataPromise
+  }
+
+  _createIntervalPromise ({id}) {
+    return new Promise(resolve => {
+      this._interval = setInterval(() => {
```

Comment:
It's not better if this._interval is initialized in the constructor, equal to null